src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -387,30 +387,37 @@ private void AddAnonymousTypeName(INamedTypeSymbol symbol)
 
         private void AddTupleTypeName(INamedTypeSymbol symbol)
         {
-            // TODO: revise to generate user-friendly name 
-            var members = string.Join(", ", symbol.GetMembers().OfType<IFieldSymbol>().Select(CreateTupleTypeMember));
+            var tupleType = (TupleTypeSymbol)symbol;
```

Comment:
Not sure the proper way to handle this. Currently only `TupleTypeSymbol` is a tuple type. I'm just going to bail out (not print a name if the symbol is cannot cast).